oggm/core/climate_prepro.py

```diff
@@ -0,0 +1,122 @@
+        if not ('gcm_temp_file' in cfg.PATHS):
+            raise ValueError("Need to set cfg.PATHS['gcm_temp_file']")
+        fpath_temp = cfg.PATHS['gcm_temp_file']
+    if fpath_precc is None:
+        if not ('gcm_precc_file' in cfg.PATHS):
+            raise ValueError("Need to set cfg.PATHS['gcm_precc_file']")
+        fpath_precc = cfg.PATHS['gcm_precc_file']
+    if fpath_precl is None:
+        if not ('gcm_precl_file' in cfg.PATHS):
+            raise ValueError("Need to set cfg.PATHS['gcm_precl_file']")
+        fpath_precl = cfg.PATHS['gcm_precl_file']
+
+    # read the files
+    with warnings.catch_warnings():
+        # Long time series are currently a pain pandas
+        warnings.filterwarnings("ignore", message='Unable to decode time axis')
+        tempds = xr.open_dataset(fpath_temp)
+    precpcds = xr.open_dataset(fpath_precc, decode_times=False)
+    preclpds = xr.open_dataset(fpath_precl, decode_times=False)
+
+    # select for location
+    lon = gdir.cenlon
+    lat = gdir.cenlat
+
+    # CESM files are in 0-360
+    if lon <= 0:
+        lon += 360
+
+    # take the closest
+    # TODO: consider GCM interpolation?
+    temp = tempds.TREFHT.sel(lat=lat, lon=lon, method='nearest')
+    precp = (precpcds.PRECC.sel(lat=lat, lon=lon, method='nearest') +
+             preclpds.PRECL.sel(lat=lat, lon=lon, method='nearest'))
+
+    temp.lon.values = temp.lon if temp.lon <= 180 else temp.lon - 360
+    precp.lon.values = precp.lon if precp.lon <= 180 else precp.lon - 360
+
+    # from normal years to hydrological years
+    sm = cfg.PARAMS['hydro_month_' + gdir.hemisphere]
+    em = sm - 1 if (sm > 1) else 12
+    # TODO: we don't check if the files actually start in January but we should
+    precp = precp[sm-1:sm-13].load()
+    temp = temp[sm-1:sm-13].load()
+    y0 = int(temp.time.values[0].strftime('%Y'))
+    y1 = int(temp.time.values[-1].strftime('%Y'))
+    time = pd.period_range('{}-{:02d}'.format(y0, sm),
+                           '{}-{:02d}'.format(y1, em), freq='M')
+
+    temp['time'] = time
+    precp['time'] = time
+    # Workaround for https://github.com/pydata/xarray/issues/1565
+    temp['month'] = ('time', time.month)
+    precp['month'] = ('time', time.month)
+    temp['year'] = ('time', time.year)
+    precp['year'] = ('time', time.year)
+    ny, r = divmod(len(time), 12)
+    assert r == 0
+
+    # Convert m s-1 to mm mth-1
+    ndays = np.tile(np.roll(cfg.DAYS_IN_MONTH, 13-sm), y1 - y0)
+    precp = precp * ndays * (60 * 60 * 24 * 1000)
+
+    # load dates in right format to save
+    dsindex = salem.GeoNetcdf(fpath_temp, monthbegin=True)
+    time1 = dsindex.variables['time']
+    time2 = time1[sm-1:sm-13] - ndays  # to hydrological years
+    time2 = netCDF4.num2date(time2, time1.units, calendar='noleap')
+    time_unit = time1.units
+
+    return process_gcm_data(gdir, filesuffix=filesuffix, precp=precp,
+                            temp=temp, time_unit=time_unit, time2=time2)
```

Comment:
I don't think we should return anything here (`process_gcm_data` doesn't have a return values).

In all cases, you should not return from the function *before* closing the files. I would recommend to shift the lines below to above, and the last call of the function should be `process_gcm_data(...)`